lib/src/http/common/util.dart

```diff
@@ -17,7 +17,14 @@ library w_transport.src.http.common.util;
 import 'dart:async';
 import 'dart:convert';
 
-StreamTransformer decodeAttempt(Encoding encoding) {
+StreamTransformer decodeAttempt(Encoding encoding) =>
+    _attemptToApplyEncoding(encoding, decode: true);
+
+StreamTransformer encodeAttempt(Encoding encoding) =>
+    _attemptToApplyEncoding(encoding, encode: true);
+
+StreamTransformer _attemptToApplyEncoding(Encoding encoding,
+    {bool encode: false, bool decode: false}) {
```

Comment:
Agreed, but it's private and I don't intend for this to be used anywhere but in this file. I added it as a way to consolidate the code for encoding & decoding since it was so similar. The `decodeAttempt` and `encodeAttempt` methods are the public API